src/main/java/com/vertispan/webdriver/gwt/gwtdriver/client/SeleniumExporter.java

```diff
@@ -84,23 +87,73 @@ public String getContainingWidgetClass(Element elt) {
 
     @Method("getContainingWidgetEltOfType")
     public Element getContainingWidgetEltOfType(Element elt, String type) {
+      Widget w = findContainingWidget(elt);
+      while (w != null && !isOfType(type, w)) {
+        w = w.getParent();
+      }
+      return w == null ? null : w.getElement();
+    }
+
+    @Method("findDescendantWidgetElementsOfType")
+    public JsArray<Element> findDescendantWidgetElementsOfType(Element elt, String type) {
+      JsArray<Element> result = JsArray.createArray().cast();
+      final Widget rootWidget = findContainingWidget(elt);
+      if (rootWidget == null) {
+        return result;
+      }
+
+      Deque<Widget> nodesToVisit = new ArrayDeque<>();
+      nodesToVisit.push(rootWidget);
```

Comment:
Not yet certain if I want the root widget to be returned.    It may make sense to ONLY pass back descendants rather than the entire tree.